### PR TITLE
Add bash qol functions to base image (and make callable from host)

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

Adding some quality of life features to our images
Test base image available at public.ecr.aws/shadowrobot/build-tools:focal-noetic-test-qol
Can be called on host with
```
bash <(curl -Ls https://raw.githubusercontent.com/shadow-robot/sr-build-tools/F_add_useful_bash_stuff/bin/sr_quality_of_life.sh)
```
Have registered a bit.ly at:
```
bash <(curl -Ls bit.ly/sr_qol)
```
(after merge, please update link in this, ask sw team for bw login)

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Review by dg-shadow — APPROVED** [review]

File: docker/ros/noetic/buildspec.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ env:
   variables:
     branch: default
     image: build-tools
-    tag: focal-noetic
+    tag: focal-noetic-test-qol
```
Comment (toliver) [inline docker/ros/noetic/buildspec.yml:8]: This will get reverted before we merge, right?

File: docker/ros/noetic/buildspec.yml [hunk 2]
```diff
@@ -40,6 +41,6 @@ phases:
       - docker build --build-arg toolset_branch=$branch -t $image .
       - docker tag $image:latest public.ecr.aws/shadowrobot/$image:$tag
       - docker push public.ecr.aws/shadowrobot/$image:$tag
-      - comm -13 <(aws ecr-public describe-image-tags --repository-name $image --region us-east-1 --output text | awk '{print $3}' | tr " " "\n" | sort -u) <(aws ecr-public describe-images --repository-name $image --region us-east-1 --output text | awk '{print $2}' | tr " " "\n" | sort -u) >> ids_to_delete.txt
-      - ids_to_delete=$(sed 's/^/imageDigest=/' ids_to_delete.txt)
-      - if [[ -s ids_to_delete.txt ]]; then aws ecr-public batch-delete-image --repository-name $image --region us-east-1 --image-ids $ids_to_delete --output text; fi
+      # - comm -13 <(aws ecr-public describe-image-tags --repository-name $image --region us-east-1 --output text | awk '{print $3}' | tr " " "\n" | sort -u) <(aws ecr-public describe-images --repository-name $image --region us-east-1 --output text | awk '{print $2}' | tr " " "\n" | sort -u) >> ids_to_delete.txt
```
Comment (toliver) [inline docker/ros/noetic/buildspec.yml:44]: Will this be removed?

**Review by toliver — CHANGES_REQUESTED** [review]

File: docker/ros/noetic/buildspec.yml (hunk 2 above)
Comment (carebare47) [reply docker/ros/noetic/buildspec.yml:44]: yes, the test image was pushing to a new repo that it didn't make sense to run this cleanup on

File: bin/sr_quality_of_life.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+fi
+
+if [ -z "${BASH_ONLY}" ]; then
+	BASH_ONLY=false
+fi
+
+if [[ "${BASH_ONLY}" == false ]]; then
+
+    echo "Installing and configuring additional quality-of-life tools"
+    sudo apt install -y tree highlight speedometer xsel gosu screen
```
[the remark this answers is not in the document]
Comment (carebare47) [reply bin/sr_quality_of_life.sh]: emacs wants to install 28 packages which takes up 205MB, do we want this in the base image?

**Pull request merged**